src/message/MoveMessage.js

```diff
@@ -0,0 +1,216 @@
+/* @flow strict-local */
+import React, { useState, useContext } from 'react';
+import { Text, View, Platform, Picker, TouchableOpacity, ScrollView } from 'react-native';
+import type { Node } from 'react';
+import Toast from 'react-native-simple-toast';
+import { ThemeContext, BRAND_COLOR } from '../styles';
+import type { RouteProp } from '../react-navigation';
+import * as api from '../api';
+import { Input } from '../common';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import { getAuth, getStreams, getOwnUser } from '../selectors';
+import { useSelector } from '../react-redux';
+import { showErrorAlert } from '../utils/info';
+import { Icon } from '../common/Icons';
+import type { Narrow, Message, Outbox } from '../types';
+import TopicAutocomplete from '../autocomplete/TopicAutocomplete';
+import { streamNameOfNarrow, streamNarrow } from '../utils/narrow';
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'move-message'>,
+  route: RouteProp<'move-message', {| message: Message | Outbox, messageNarrow: Narrow |}>,
+|}>;
+
+const inputMarginPadding = {
+  paddingHorizontal: 8,
+  paddingVertical: Platform.select({
+    ios: 8,
+    android: 2,
+  }),
+};
+
+export default function MoveMessage(props: Props): Node {
+  const themeContext = useContext(ThemeContext);
+  const backgroundColor = themeContext.backgroundColor;
+  const cardColor = themeContext.cardColor;
+  const iconName = Platform.OS === 'android' ? 'arrow-left' : 'chevron-left';
+  const auth = useSelector(getAuth);
+  const allStreams = useSelector(getStreams);
+  const isAdmin = useSelector(getOwnUser).is_admin;
+  const messageId = props.route.params.message.id;
+  const currentStreamName = streamNameOfNarrow(props.route.params.messageNarrow);
+  const currentStreamId = allStreams.find(s => s.name === currentStreamName)?.stream_id;
```

Comment:
Avoid linear searches like this one: #3339

That's why at https://github.com/zulip/zulip-mobile/pull/5189#discussion_r781797902 I recommended `getStreamsByName`, which is a data structure for accomplishing the same lookup efficiently.